fl4health/clients/fenda_client.py

```diff
@@ -64,19 +64,21 @@ def get_parameter_exchanger(self, config: Config) -> ParameterExchanger:
         assert isinstance(self.model, FendaModel)
         return FixedLayerExchanger(self.model.layers_to_exchange())
 
-    def predict(self, input: torch.Tensor) -> Tuple[Dict[str, torch.Tensor], Dict[str, torch.Tensor]]:
+    def predict(self, input: TorchInputType) -> Tuple[Dict[str, torch.Tensor], Dict[str, torch.Tensor]]:
         """
         Computes the prediction(s) and features of the model(s) given the input.
 
         Args:
-            input (torch.Tensor): Inputs to be fed into the model.
+            input (TorchInputType): Inputs to be fed into the model. TorchInputType is simply an alias
+            for the union of torch.Tensor and Dict[str, torch.Tensor].
 
         Returns:
             Tuple[Dict[str, torch.Tensor], Dict[str, torch.Tensor]]: A tuple in which the first element
             contains predictions indexed by name and the second element contains intermediate activations
             index by name. Specificaly the features of the model, features of the global model and features of
             the old model are returned. All predictions included in dictionary will be used to compute metrics.
         """
+        assert isinstance(input, torch.Tensor)
```

Comment:
Similar to the ensemble, do we need this assertion?

Reply:
See https://github.com/VectorInstitute/FL4Health/pull/124#discussion_r1566168462

Reply:
Also just for clarity, I did `self.model(**input)` for ditto because the model there has a general `nn.Module` type, unlike the ensemble/moon/fenda clients.